tests/unit/amo/components/TestSearch.js

```diff
@@ -433,6 +433,16 @@ describe(__filename, () => {
     expect(wrapper.find(NotFound)).toHaveLength(1);
   });
 
+  it('renders a robots meta when there is no results', () => {
+    const root = render({ count: 0 });
+
+    expect(root.find('meta[name="robots"]')).toHaveLength(1);
+    expect(root.find('meta[name="robots"]')).toHaveProp(
+      'content',
+      'noindex, follow',
+    );
+  });
+
```

Comment:
Because we really don't want SEO regressions and because it can be a little difficult to notice those regressions in time... I think we need a few more tests proving that:
- We don't accidentally add the meta to non-search pages that don't have `count` in `props`.
- We don't accidentally add the meta to search pages with a positive result count
- We add the meta to category pages specifically when they have a result count of 0
- We don't accidentally add the meta to category pages specifically with a positive result count

Reply:
Right, I can add some assertions to make sure the meta isn't present in most cases.

Reply:
> * We don't accidentally add the meta to non-search pages that don't have `count` in `props`.

Not sure what that means, this is only impacting a `Search` component so non search pages cannot be affected.

> * We don't accidentally add the meta to search pages with a positive result count

Added (assertion + test case)

> * We add the meta to category pages specifically when they have a result count of 0
> * We don't accidentally add the meta to category pages specifically with a positive result count

Category pages use the `Search` component so that's covered by the tests of this component.

Reply:
> Not sure what that means, this is only impacting a `Search` component so non search pages cannot be affected.

I read too quickly :) Still would be nice to have a test proving we handle when `count` is `undefined` (or even a falsy non-integer).

>> We add the meta to category pages specifically when they have a result count of 0
>> We don't accidentally add the meta to category pages specifically with a positive result count
>
> Category pages use the Search component so that's covered by the tests of this component.

Yes, but my point was to have some redundancy. We know we are using the Search component, but if for some reason the category page starts passing `0` as the `count` in props for instance, it would break.

Reply:
> Still would be nice to have a test proving we handle when `count` is `undefined` (or even a falsy non-integer).

This is an injected prop so unlikely to end up in a state like this but we can hack around and have a test case for that. The check in the component is a strict comparison to an integer so that won't be a problem.

> Yes, but my point was to have some redundancy. We know we are using the Search component, but if for some reason the category page starts passing `0` as the `count` in props for instance, it would break.

This isn't easily doable because Flow would prevent that (I added the prop on L94):

```
yarn flow
yarn run v1.22.10
$ flow
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/amo/pages/CategoryPage/index.js:93:10

Cannot create Search element because property count is missing in
Props [1] but exists in props [2]. [prop-missing]

     src/amo/pages/CategoryPage/index.js
      90│     return (
      91│       <Page>
      92│         <HeadLinks />
 [2]  93│         <Search
      94│           count={0}
      95│           enableSearchFilters
      96│           filters={filtersForSearch}
      97│           pageTitle={this.getPageTitle(categoryName)}
      98│           paginationQueryParams={convertFiltersToQueryParams(filters)}
      99│           pathname={getCategoryResultsPathname({
     100│             addonType,
     101│             slug: categorySlug,
     102│           })}
     103│         />

     src/amo/components/Search/index.js
 [1] 334│ const Search: React.ComponentType<Props> = compose(
```